staging/src/k8s.io/csi-api/pkg/apis/csi/v1alpha1/types.go

```diff
@@ -97,8 +106,33 @@ type CSINodeInfo struct {
 	CSIDrivers []CSIDriverInfo `json:"csiDrivers" patchStrategy:"merge" patchMergeKey:"driver"`
 }
 
-// CSIDriverInfo contains information about one CSI driver installed on a node.
+// CSIDriverInfo holds information about one CSI driver installed on a node.
 type CSIDriverInfo struct {
+	// spec is the specification of CSIDriverInfo
+	Spec CSIDriverInfoSpec `json:"spec"`
+
+	// status is the current status of CSIDriverInfo
+	Status CSIDriverInfoStatus `json:"status"`
+}
+
+// CSIDriverInfoStatus holds information about the status of one CSI driver installed on a node
+type CSIDriverInfoStatus struct {
+	// available is a boolean representing whether the driver has been installed
+	// on this node or not.
+	Available bool `json:"available"`
```

Comment:
OK, now that I understand the next field, does this field serve any additional purpose? Do we need both?

Reply:
yes, the driver can be in one of 4 states. 

- available & in-tree mechanism
- available & csi mechanism
- not available & in-tree mechanism
- not available & csi mechanism

and the Attach/Detach controller will behave differently when the driver is in each of those states.

Reply:
What's the difference between (not available & in-tree mechanism) and (not available & csi mechanism)? In what case would a CSIDriverInfo be unavailable and still be included in the "list of CSI drivers running on the node"?

Reply:
Can you answer Mike's question?

Reply:
bump

Reply:
1) 
- Not available & In-tree mechanism: Attach Detach controller attaches/detaches volumes with the in-tree mechanism
- Not available & CSI Mechanism: Attach Detach controller throws an error (CSI Driver not available). 
and for completeness
- Available & In-tree Mechanism: Attach Detach controller attaches/detaches volumes with the in-tree mechanism
- Available & CSI Mechanism: Attach Detach controller attaches/detaches volumes with the csi mechanism

2) There will be a followup PR that adds "well known" CSI Drivers to the CSINodeInfo on node startup as `unavailable`